// ngx-scrolltop/esm2022/lib/ngx-scrolltop.core.service.mjs
import { DOCUMENT } from '@angular/common';
import { Inject, Injectable } from '@angular/core';
import { polyfill as smoothscrollPolyfill } from 'seamless-scroll-polyfill';
import * as i0 from "@angular/core";
export class NgxScrollTopCoreService {
    constructor(document) {
        this.document = document;
        this.scrolledFromTop = false;
        this.isBrowser = typeof window !== 'undefined';
        this.alreadyActivated = false;
    }
    onWindowScroll(mode) {
        const position = this.document.documentElement?.scrollTop || this.document.scrollingElement?.scrollTop;
        switch (mode) {
            case 'classic':
                return this.classicMode(position);
            case 'smart':
                return this.smartMode(position);
        }
    }
    classicMode(position) {
        if (this.isBrowser && position > window.innerHeight) {
            return true;
        }
        else {
            return false;
        }
    }
    smartMode(position) {
        let show = false;
        if (position === 0) {
            show = false;
            this.scrolledFromTop = false;
        }
        if (this.scrolledFromTop && this.scrollOffset > position) {
            show = true;
        }
        if (this.isBrowser && position > window.innerHeight * 2) {
            this.scrolledFromTop = true;
            this.scrollOffset = position;
        }
        return show;
    }
    scrollToTop() {
        if (this.isBrowser) {
            // Kick off the polyfill for iOS Safari
            if (!this.alreadyActivated) {
                smoothscrollPolyfill();
                this.alreadyActivated = true;
            }
            // Scroll to the top
            window.scroll({ top: 0, left: 0, behavior: 'smooth' });
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopCoreService, deps: [{ token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopCoreService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopCoreService, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }] });
