// ngx-scrolltop/esm2022/lib/ngx-scrolltop.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxScrollTopComponent } from './ngx-scrolltop.component';
import { NgxScrollTopDirective } from './ngx-scrolltop.directive';
import { NgxScrollTopCoreService } from './ngx-scrolltop.core.service';
import * as i0 from "@angular/core";
export class NgxScrollTopModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopModule, declarations: [NgxScrollTopComponent, NgxScrollTopDirective], imports: [CommonModule], exports: [NgxScrollTopComponent, NgxScrollTopDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopModule, providers: [NgxScrollTopCoreService], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxScrollTopComponent, NgxScrollTopDirective],
                    imports: [CommonModule],
                    providers: [NgxScrollTopCoreService],
                    exports: [NgxScrollTopComponent, NgxScrollTopDirective],
                }]
        }] });
