// ngx-scrolltop/esm2022/lib/ngx-scrolltop.directive.mjs
import { Directive, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-scrolltop.core.service";
export class NgxScrollTopDirective {
    constructor(el, core) {
        this.el = el;
        this.core = core;
        this.mode = 'classic';
        this.show = false;
        this.hideElement();
    }
    onWindowScroll() {
        const show = this.core.onWindowScroll(this.mode);
        // Performance boost. Only update the DOM when the state changes.
        if (this.show !== show) {
            show ? this.showElement() : this.hideElement();
            this.show = show;
        }
    }
    onClick() {
        this.scrollToTop();
    }
    hideElement() {
        this.el.nativeElement.style.display = 'none';
    }
    showElement() {
        this.el.nativeElement.style.display = '';
    }
    scrollToTop() {
        this.core.scrollToTop();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopDirective, deps: [{ token: i0.ElementRef }, { token: i1.NgxScrollTopCoreService }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.0.6", type: NgxScrollTopDirective, selector: "[ngxScrollTop]", inputs: { mode: ["ngxScrollTopMode", "mode"] }, host: { listeners: { "window:scroll": "onWindowScroll()", "click": "onClick()" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.6", ngImport: i0, type: NgxScrollTopDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[ngxScrollTop]',
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i1.NgxScrollTopCoreService }], propDecorators: { mode: [{
                type: Input,
                args: ['ngxScrollTopMode']
            }], onWindowScroll: [{
                type: HostListener,
                args: ['window:scroll']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
